pkg/component/dependencywatchdog/access.go

```diff
@@ -49,7 +58,7 @@ func NewAccess(
 	namespace string,
 	secretsManager secretsmanager.Interface,
 	values AccessValues,
-) component.Deployer {
+) *dependencyWatchdogAccess {
```

Comment:
An exposed function should ideally not return un-exposed type. Golang as a language allows it but it is not considered a good practice. Instead one can easily use interface embedding.
```go
type DWDAccess interface {
	component.Deployer
	DeployMigrate(ctx context.Context) error
}
```
and then have `NewAccess` return `DWDAccess` instead.